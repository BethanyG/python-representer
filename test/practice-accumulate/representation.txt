Module([FunctionDef('placeholder_0', arguments([], [arg('placeholder_1', lineno=4, col_offset=15, end_lineno=4, end_col_offset=25), arg('placeholder_2', lineno=4, col_offset=27, end_lineno=4, end_col_offset=36)], kwonlyargs=[], kw_defaults=[], defaults=[]), [Assign([Name('placeholder_3', Store(), lineno=5, col_offset=4, end_lineno=5, end_col_offset=12)], List([], Load(), lineno=5, col_offset=15, end_lineno=5, end_col_offset=17), lineno=5, col_offset=4, end_lineno=5, end_col_offset=17), For(Name('placeholder_4', Store(), lineno=6, col_offset=8, end_lineno=6, end_col_offset=15), Name('placeholder_1', Load(), lineno=6, col_offset=19, end_lineno=6, end_col_offset=29), [Expr(Call(Attribute(Name('placeholder_3', Load(), lineno=7, col_offset=8, end_lineno=7, end_col_offset=16), 'append', Load(), lineno=7, col_offset=8, end_lineno=7, end_col_offset=23), [Call(Name('placeholder_2', Load(), lineno=7, col_offset=24, end_lineno=7, end_col_offset=33), [Name('placeholder_4', Load(), lineno=7, col_offset=34, end_lineno=7, end_col_offset=41)], [], lineno=7, col_offset=24, end_lineno=7, end_col_offset=42)], [], lineno=7, col_offset=8, end_lineno=7, end_col_offset=43), lineno=7, col_offset=8, end_lineno=7, end_col_offset=43)], [], lineno=6, col_offset=4, end_lineno=7, end_col_offset=43), Return(Name('placeholder_3', Load(), lineno=8, col_offset=11, end_lineno=8, end_col_offset=19), lineno=8, col_offset=4, end_lineno=8, end_col_offset=19)], [], lineno=4, col_offset=0, end_lineno=8, end_col_offset=19)], [])