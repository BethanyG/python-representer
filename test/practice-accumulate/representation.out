## BEGIN NORMALIZED CODE ##
def placeholder_0(placeholder_1, placeholder_2):
    placeholder_3 = []
    for placeholder_4 in placeholder_1:
        placeholder_3.append(placeholder_2(placeholder_4))
    return placeholder_3

## END NORMALIZED CODE ##

## BEGIN TREE BEFORE ##
Module(
  [
    FunctionDef(
      'accumulate',
      arguments(
        [],
        [
          arg(
            'collection',
            lineno=4,
            col_offset=15,
            end_lineno=4,
            end_col_offset=25),
          arg(
            'operation',
            lineno=4,
            col_offset=27,
            end_lineno=4,
            end_col_offset=36)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Assign(
          [
            Name(
              'response',
              Store(),
              lineno=5,
              col_offset=4,
              end_lineno=5,
              end_col_offset=12)],
          List(
            [],
            Load(),
            lineno=5,
            col_offset=15,
            end_lineno=5,
            end_col_offset=17),
          lineno=5,
          col_offset=4,
          end_lineno=5,
          end_col_offset=17),
        For(
          Name(
            'element',
            Store(),
            lineno=6,
            col_offset=8,
            end_lineno=6,
            end_col_offset=15),
          Name(
            'collection',
            Load(),
            lineno=6,
            col_offset=19,
            end_lineno=6,
            end_col_offset=29),
          [
            Expr(
              Call(
                Attribute(
                  Name(
                    'response',
                    Load(),
                    lineno=7,
                    col_offset=8,
                    end_lineno=7,
                    end_col_offset=16),
                  'append',
                  Load(),
                  lineno=7,
                  col_offset=8,
                  end_lineno=7,
                  end_col_offset=23),
                [
                  Call(
                    Name(
                      'operation',
                      Load(),
                      lineno=7,
                      col_offset=24,
                      end_lineno=7,
                      end_col_offset=33),
                    [
                      Name(
                        'element',
                        Load(),
                        lineno=7,
                        col_offset=34,
                        end_lineno=7,
                        end_col_offset=41)],
                    [],
                    lineno=7,
                    col_offset=24,
                    end_lineno=7,
                    end_col_offset=42)],
                [],
                lineno=7,
                col_offset=8,
                end_lineno=7,
                end_col_offset=43),
              lineno=7,
              col_offset=8,
              end_lineno=7,
              end_col_offset=43)],
          [],
          lineno=6,
          col_offset=4,
          end_lineno=7,
          end_col_offset=43),
        Return(
          Name(
            'response',
            Load(),
            lineno=8,
            col_offset=11,
            end_lineno=8,
            end_col_offset=19),
          lineno=8,
          col_offset=4,
          end_lineno=8,
          end_col_offset=19)],
      [],
      lineno=4,
      col_offset=0,
      end_lineno=8,
      end_col_offset=19)],
  [])
## END TREE BEFORE ##

## BEGIN NORMALIZED TREE ##
Module(
  [
    FunctionDef(
      'placeholder_0',
      arguments(
        [],
        [
          arg(
            'placeholder_1',
            lineno=4,
            col_offset=15,
            end_lineno=4,
            end_col_offset=25),
          arg(
            'placeholder_2',
            lineno=4,
            col_offset=27,
            end_lineno=4,
            end_col_offset=36)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Assign(
          [
            Name(
              'placeholder_3',
              Store(),
              lineno=5,
              col_offset=4,
              end_lineno=5,
              end_col_offset=12)],
          List(
            [],
            Load(),
            lineno=5,
            col_offset=15,
            end_lineno=5,
            end_col_offset=17),
          lineno=5,
          col_offset=4,
          end_lineno=5,
          end_col_offset=17),
        For(
          Name(
            'placeholder_4',
            Store(),
            lineno=6,
            col_offset=8,
            end_lineno=6,
            end_col_offset=15),
          Name(
            'placeholder_1',
            Load(),
            lineno=6,
            col_offset=19,
            end_lineno=6,
            end_col_offset=29),
          [
            Expr(
              Call(
                Attribute(
                  Name(
                    'placeholder_3',
                    Load(),
                    lineno=7,
                    col_offset=8,
                    end_lineno=7,
                    end_col_offset=16),
                  'append',
                  Load(),
                  lineno=7,
                  col_offset=8,
                  end_lineno=7,
                  end_col_offset=23),
                [
                  Call(
                    Name(
                      'placeholder_2',
                      Load(),
                      lineno=7,
                      col_offset=24,
                      end_lineno=7,
                      end_col_offset=33),
                    [
                      Name(
                        'placeholder_4',
                        Load(),
                        lineno=7,
                        col_offset=34,
                        end_lineno=7,
                        end_col_offset=41)],
                    [],
                    lineno=7,
                    col_offset=24,
                    end_lineno=7,
                    end_col_offset=42)],
                [],
                lineno=7,
                col_offset=8,
                end_lineno=7,
                end_col_offset=43),
              lineno=7,
              col_offset=8,
              end_lineno=7,
              end_col_offset=43)],
          [],
          lineno=6,
          col_offset=4,
          end_lineno=7,
          end_col_offset=43),
        Return(
          Name(
            'placeholder_3',
            Load(),
            lineno=8,
            col_offset=11,
            end_lineno=8,
            end_col_offset=19),
          lineno=8,
          col_offset=4,
          end_lineno=8,
          end_col_offset=19)],
      [],
      lineno=4,
      col_offset=0,
      end_lineno=8,
      end_col_offset=19)],
  [])
## END NORMALIZED TREE ##